### PR TITLE
Workflow Best Practices naming conventions draft

### PR DESCRIPTION
**Pull request by tshtatland**
Add Galaxy Workflow Best Practices naming conventions draft.
Intended audience: internal and external Galaxy workflow developers, maintainers, and users.

- The style is easy for the reader and maintainer.
- Look and feel is consistent with a very popular, influential and respected book "Perl Best Practices" by Damian Conway.

Add Galaxy Workflow Best Practices to README

- Short description: what they are and what they are not. Hopefully, should help prevent flame wars.
- Add references and sources.

### CODE REVIEW

**Comment by bgruening** [comment]
Cool thanks @tshtatland for working on this!

**Review by bwlang — APPROVED** [review]
I just found this PR - it's old but I think this is still helpful

File: README.md [hunk 1]
```diff
@@ -1,5 +1,81 @@
 # IWC - Intergalactic Workflow Commission
 
+## This group exists to create a registry of community workflows.
```
Comment (mvdbeek) [inline README.md:3]: I like the current readme more, it is short and to the point.

File: README.md [hunk 2]
```diff
@@ -1,5 +1,81 @@
 # IWC - Intergalactic Workflow Commission
 
+## This group exists to create a registry of community workflows. 
+
+### Reproducibility is important. Our goals are to:
+ - foster workflow use
+ - incorporate versioning
+ - capture more metadata: (names, versions, authors, use cases, etc.)
+ - help scientists find workflows!
```
Comment (mvdbeek) [inline README.md:9]: That's redundant with the contributing document.

File: README.md [hunk 3]
```diff
@@ -1,5 +1,81 @@
...
+
+### Methods Proposal
+ - discover workflows
+ - establish quality standards (version numbers, tools exist, author is set, training available, test exists ...)
+ - automated testing of workflows utilizing [Planemo](https://github.com/galaxyproject/planemo) test
+ - two layers of test data:
+   - basic functionality testing (quick to run, tools exist) 
+   - downsampled data (validity of workflow results)
```
Comment (mvdbeek) [inline README.md:17]: I agree that we should have some example analysis with the full details, but we're not there yet. The rest is again redundant with the contributing document

File: README.md [hunk 4]
```diff
@@ -1,5 +1,81 @@
...
+
+### Import process:
+ - submit a pull request (PR) that adds a link to a GA file and metadata  
+ - confirm that the GA and metadata conform to standards (linting criteria to be defined)
+ - test the workflow 
+ - if linting passes, tests pass, and human review passes, PR is merged and added to registry list
```
Comment (mvdbeek) [inline README.md:23]: it's all in the contributing document

File: README.md [hunk 5]
```diff
@@ -1,5 +1,81 @@
...
+
+### Meeting minutes:
+ * [the notes](https://github.com/galaxyproject/iwc/issues/3) from the initial meeting at GCCBOSC2018 CoFest
```
Comment (mvdbeek) [inline README.md:26]: meeting minutes are attached to the google calendar, linking them up would be a good idea.

File: README.md [hunk 6]
```diff
@@ -1,5 +1,81 @@
...
+
+## Galaxy Workflow Best Practices
```
Comment (mvdbeek) [inline README.md:28]: This section is partially redundant with https://planemo.readthedocs.io/en/latest/best_practices_workflows.html which we link to, and partially not relevant.

File: best_practices/01_reference_guide.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
+## Naming Conventions
```
Comment (mvdbeek) [inline best_practices/01_reference_guide.md:1]: Good to have this, but these are not our current naming conventions, they are however outlined in the contributing document

Comment (mvdbeek) [reply best_practices/01_reference_guide.md:1]: Overall it's unclear who the audience is for this document.

File: best_practices/naming_conventions.md [hunk 8]
```diff
@@ -0,0 +1,273 @@
+# Naming Conventions
```
Comment (mvdbeek) [inline best_practices/naming_conventions.md:1]: This is way too long, most of it is enforced anyway by github conventions.

File: best_practices/running_workflows.md [hunk 9]
```diff
@@ -0,0 +1,126 @@
+# Running workflows
+
+*"I ran your workflow, but it does not work"*
+
+\-- Anonymous
+
+## Record-keeping
+### Keep records of the workflows, inputs and parameters.
```
Comment (mvdbeek) [inline best_practices/running_workflows.md:8]: All of that is recorded in Galaxy, does this maybe date from many years back when it wasn't in the user interface ?

File: best_practices/workflows_misc.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
+# Workflows - Miscellaneous
+
+
+## Citing
```
Comment (mvdbeek) [inline best_practices/workflows_misc.md:4]: Citation information is embedded everywhere, in the Galaxy workflow, in dockstore/workflowhub etc, I don't think this is necessary to repeat here.

**Review by mvdbeek — CHANGES_REQUESTED** [review]
Thanks @tshtatland @bgruening,

I like the idea of providing full analysis details, we could start with a public history of an example run, however I wouldn't make that mandatory. Putting the meeting notes and calendar in a more public spot is also a good idea. Otherwise I think that the contributing document is not too bad for getting started. I would say adding relevant bits to the existing documentation is probably the right way to do ? 

I hope I'll see you at the next IWC meeting!

**Pull request closed without merging**